### PR TITLE
MSEARCH-696: Make all searches case-insensitive

### PR DESCRIPTION
**Pull request by TsaghikKhachatryan**
### Purpose
Make Inventory and Authority searches case-insensitive.

### Approach
Change the field types of "keyword" to "keyword_lowercase".

Create a custom analyzer named "whitespace_analyzer" with the "lowercase" filter and add it as an analyzer to the existing "whitespace" indices.

### Changes Checklist
- [ ] **API Changes**: Document any API paths, methods, request, or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [ ] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [ ] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
https://folio-org.atlassian.net/browse/MSEARCH-696
https://folio-org.atlassian.net/browse/MSEARCH-688

### CODE REVIEW

File: src/main/java/org/folio/search/utils/SearchUtils.java [hunk 1]
```diff
@@ -307,6 +307,16 @@ public static boolean isEmptyString(Object value) {
     return value instanceof String string && string.isEmpty();
   }
 
+  /**
+   * Check if passed value is blank.
+   *
+   * @param value - value to check
+   * @return true - if value is blank, false - otherwise
+   */
+  private static boolean isBlankString(String value) {
```
Comment (psmagin) [inline src/main/java/org/folio/search/utils/SearchUtils.java:316]: I don't think that this method makes sense, as there no other logic apart from using string utils method

File: src/main/resources/elasticsearch/index/instance.json [hunk 2]
```diff
@@ -33,6 +33,11 @@
         "filter": [ "folio_word_delimiter_graph", "icu_folding" ],
         "char_filter": [ "and_char_replacement" ],
         "type": "custom"
+      },
+      "whitespace_analyzer": {
```
Comment (psmagin) [inline src/main/resources/elasticsearch/index/instance.json:37]: whitespace_lowercase probably will make it consistent with other

File: src/main/resources/model/authority.json [hunk 3]
```diff
@@ -7,11 +7,11 @@
   },
   "fields": {
     "id": {
-      "index": "keyword",
+      "index": "keyword_lowercase",
       "showInResponse": [ "search", "browse" ]
     },
     "tenantId": {
-      "index": "keyword",
+      "index": "keyword_lowercase",
```
Comment (psmagin) [inline src/main/resources/model/authority.json:14]: Please back it to just keyword as well as other tenantId fields

File: src/main/resources/model/authority.json [hunk 4]
```diff
@@ -323,8 +323,9 @@
       "default": "NULL"
     },
     "naturalId": {
-      "index": "keyword",
-      "showInResponse": [ "search", "browse" ]
+      "index": "keyword_lowercase",
+      "showInResponse": [ "search", "browse" ],
+      "searchTermProcessor": "naturalIdSearchTermProcessor"
     },
```
Comment (psmagin) [inline src/main/resources/model/authority.json:329]: Why there a need for processor that makes value lowecase if we already use keyword-lowercase?

**Review by psmagin — CHANGES_REQUESTED** [review]

File: src/main/resources/elasticsearch/index/instance.json (hunk 2 above)
Comment (TsaghikKhachatryan) [reply src/main/resources/elasticsearch/index/instance.json:37]: Are you asking if it is consistent with  keyword_lowercase? I guess, that the distinction is that "whitespace" enables full-text searches, whereas "keyword" makes exact matches. Originally, "whitespace" was used for identifiers.value. Therefore, I included an additional analyzer to preserve the original functionality

Comment (psmagin) [reply src/main/resources/elasticsearch/index/instance.json:37]: I mean probably will be better to rename it to whitespace_lowercase_analyzer

**Review by psmagin — APPROVED** [review]

**Review by viacheslavpoliakov — APPROVED** [review]

**Pull request merged**